systemtest/src/test/java/io/strimzi/systemtest/metrics/MetricsST.java

```diff
@@ -329,6 +333,26 @@ void testKafkaBridgeMetrics() {
         assertThat(values.stream().mapToDouble(i -> i).sum(), is((double) 1));
     }
 
+    @Test
+    @Tag(CRUISE_CONTROL)
+    void testCruiseControlMetrics() {
+
+        String cruiseControlMetrics = CruiseControlUtils.callApi(CruiseControlUtils.SupportedHttpMethods.GET, "/metrics");
+
+        Matcher regex = Pattern.compile("^([^#].*)\\s+([^\\s]*)$", Pattern.MULTILINE).matcher(cruiseControlMetrics);
+
+        while (regex.find()) {
```

Comment:
Might we want to assert that we pass through this more than 0 times? If the interface changes the regex will immediately not find and this test becomes a no-op if I am not mistaken.

Reply:
Yeah, that's correct consideration. What about `assertThat(regex.end(), greaterThan(0));` ?

Reply:
Changed to 
```
LOGGER.info("Verifying that there is already more than 100 different metrics");
assertThat(regex.results().count(), greaterThan(100L));
```

Reply:
That looks good to me, whilst variable sized checks are useful for extensibility, we should add the sanity check assert that you suggested to check that we have an expected sized sample

Reply:
hmmm `regex.results().count()` is only for Java 11 :(, This one `regex.end()` also doens't work. So maybe i can assert that line will have 2 groups.

Reply:
WDYT  @samuel-hawker

Reply:
Honestly, I don't have a strong opinion on this, I would suggest the cleaner Java 11 check you leave in, but commented out, and when we stop supporting java 8 we remove the old assertion.